rclcpp/src/rclcpp/node_interfaces/node_base.cpp

```diff
@@ -65,10 +65,13 @@ NodeBase::NodeBase(
   // Create the rcl node and store it in a shared_ptr with a custom destructor.
   std::unique_ptr<rcl_node_t> rcl_node(new rcl_node_t(rcl_get_zero_initialized_node()));
 
-  ret = rcl_node_init(
-    rcl_node.get(),
-    node_name.c_str(), namespace_.c_str(),
-    context_->get_rcl_context().get(), &rcl_node_options);
+  {
+    std::lock_guard<std::mutex> guard(*(context_->get_rcl_logging_configure_mutex()));
```

Comment:
There are two subtle bugs here due to using a shared_ptr:

1. What if it's null?
2. What if the object is deleted after the shared_ptr is deterrence?

Reply:
> What if it's null?

We should add a check and handle that case correctly.

> What if the object is deleted after the shared_ptr is deterrence?

If the `Node` holds the shared pointer, that's avoided.